src/Renci.SshNet/Sftp/SftpFileReader.cs

```diff
@@ -53,12 +53,14 @@ internal class SftpFileReader : ISftpFileReader
         /// <param name="chunkSize">The size of a individual read-ahead chunk.</param>
         /// <param name="maxPendingReads">The maximum number of pending reads.</param>
         /// <param name="fileSize">The size of the file, if known; otherwise, <c>null</c>.</param>
-        public SftpFileReader(byte[] handle, ISftpSession sftpSession, uint chunkSize, int maxPendingReads, long? fileSize)
+        /// <param name="offset">The offset to resume from</param>
+        public SftpFileReader(byte[] handle, ISftpSession sftpSession, uint chunkSize, int maxPendingReads, long? fileSize, ulong offset = 0)
         {
             _handle = handle;
             _sftpSession = sftpSession;
             _chunkSize = chunkSize;
             _fileSize = fileSize;
+            _offset = _readAheadOffset = offset;
```

Comment:
File download: set the initial offset of the remote file to the same offset as the local output file. Read-ahead also starts at the same offset